src/orient/rational-calc.orient

```diff
@@ -0,0 +1,354 @@
+    describe(window_column_leaves, "Constraints to check all the leaves generation required for all windows")
+
+    // D
+    // Data layer committed with merkle tree
+    data_inclusion = merkle_tree_datahash_constraints * log2(nodes)
+    data_inclusions = data_inclusion * porep_challenges
+    window_data_inclusions = data_inclusion * porep_challenges * windows
+
+    // C
+    window_inclusion = merkle_tree_hash_constraints * log2(window_nodes)
+    window_inclusions =  window_inclusion * porep_challenges * (graph_parents + 1)
+    column_leaf = column_leaf_hash_constraints * drg_layers
+    column_leaves = column_leaf * porep_challenges * (graph_parents + 1)
+    window_column_leaves = column_leaves * windows
+    labeling_proof = kdf_constraints * graph_parents
+    labeling_proofs = labeling_proof * drg_layers * porep_challenges
+    window_labeling_proof = labeling_proofs * windows
+
+    // Q and R
+    replica_inclusion = merkle_tree_hash_constraints * log2(nodes)
+    replica_inclusions = replica_inclusion * porep_challenges
+    window_replica_inclusions = replica_inclusions * windows
+
+    porep_snark_partitions = porep_snark_constraints / porep_snark_partition_constraints
+    porep_snark_size = porep_snark_partitions * snark_size
+    porep_snark_size_kb = porep_snark_size / 1024
+
+    Constraints (WindowEncoding):
+      porep_snark_constraints = window_inclusions + window_column_leaves + window_labeling_proof + window_data_inclusions + window_replica_inclusions*2
+
+    Constraints (SectorEncoding):
+      porep_snark_constraints = window_inclusions + window_column_leaves + window_labeling_proof + data_inclusions + replica_inclusions*2
+
+    Constraints (Wrapping):
+      porep_snark_constraints = window_inclusions + window_column_leaves + window_labeling_proof + data_inclusions + replica_inclusions*2
+
+    // Column comitting data or replica. 
+
+Rationality:
+  describe(breakeven_nodes, "number of nodes / sector / challenge period after which it's cheaper to store the entire sector, than re-encode those")
+  describe(polling_time, "Time between two PoSt proofs")
+  describe(challenge_periods_per_month, "Number of challenge periods in a month")
+  describe(regeneration_fraction, "Fraction of nodes of a sector for which re-encoding them every `polling_time` costs as much as storing them")
+
+  breakeven_nodes = ((cost_gb_per_month *  sector_size_gb) / (cost_node_encoding * challenge_periods_per_month))
+  challenge_periods_per_month = secs_in_month / polling_time
+  regeneration_fraction = breakeven_nodes / nodes
+
+  Costs:
+    Compute:
+      describe(cost_node_encoding, "Cost of encoding a single node")
+      describe(nodes_per_second, "Nodes encoded per second using the fastest hardware")
+      describe(nodes_per_year, "Nodes encoded per year using the fastest hardware")
+      describe(hash_gb_per_second, "GiB that can be hashed per second")
```

Comment:
Check to make sure the source data you are using is actually in GiB rather than GB — or else provide a conversion. Or maybe you mean GB, as in the variable name. Either way, make the description and name consistent.

Reply:
fixed in latest